src/zenml/config/step_configurations.py

```diff
@@ -54,25 +60,30 @@ class PartialArtifactConfiguration(StrictBaseModel):
     # for all steps/outputs
     default_materializer_source: Optional[Source] = None
 
-    @root_validator(pre=True)
+    @model_validator(mode="before")
```

Comment:
Why not `after`?

Reply:
This is the case for both the old and the new validator here: It can take values that are not actually a part of the model. The validation here happens first and the processed result is passed to the actual pydantic validation.